adapters/net_http/lib/opentelemetry/adapters/net/http/patches/instrumentation.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Net
+      module HTTP
+        module Patches
+          # Module to prepend to Net::HTTP for instrumentation
+          module Instrumentation
+            USE_SSL_TO_SCHEME = { false => 'http', true => 'https' }.freeze
+
+            def request(req, body = nil, &block) # rubocop:disable Metrics/MethodLength
+              tracer.in_span(
+                "HTTP #{req.method}",
```

Comment:
Given this naming convention, we could cache the span name in a map with method as the key, and  `HTTP {method}` as the value.

Reply:
@mwear Done. Is there a way you would like to provide for a fallback in case the lookup returns `nil`?

Reply:
You could make the cache adaptive: `Hash.new do |h, k| { h[k] = "HTTP #{k}" }`.